simba/report.py

```diff
@@ -243,6 +254,19 @@ def generate(schedule, scenario, args):
             t = sim_start_time + i * scenario.interval
             socs = [str(rotation_socs[k][i]) for k in rotations]
             data.append([str(t)] + socs)
+
+        # add active rotations column to rotation_socs.csv
+        active_vehicles = [len(scenario.components.vehicles)] * scenario.n_intervals
+        depot_stations = [station for station in scenario.components.grid_connectors
+                          if schedule.stations[station]["type"] == "deps"]
+        for station_name in depot_stations:
+            station_ts = scenario.connChargeByTS[station_name]
+            for index, step in enumerate(station_ts):
+                active_vehicles[index] -= len(step)
+        data[0].append("# active_rotations")
+        for i in range(len(data)):
+            data[i].append(active_vehicles[i-1])
```

Comment:
Needs more comments what is happening. What are active_vehicles? As of now, this takes all vehicles of the simulation and then subtracts the number of charging stations at depots? Not even the active charging stations, but the total count? It does not help that the column name is active_rotations while the variable name is active_vehicles. 
I think what you are trying to do is counting vehicles that are not currently in a depot (i.e., on a rotation). However, the mapping of what vehicle is connected where is not really an information that SpiceEV exposes (it is known for the strategy during runtime, but not after the strategy finishes). I think it would be more straightforward to use the information SimBA has on hand, e.g. rotation data. Pseudocode:
```
active_rotations = 0 for each timestep
for each rotation:
    add one vehicle for each timestep of the rotation to active_rotations
```